google/resource_cloudiot_registry.go

```diff
@@ -0,0 +1,411 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"google.golang.org/api/cloudiot/v1"
+	"strings"
+)
+
+const (
+	mqttStateUnspecified  = "MQTT_STATE_UNSPECIFIED"
+	mqttEnabled           = "MQTT_ENABLED"
+	mqttDisabled          = "MQTT_DISABLED"
+	httpStateUnspecified  = "HTTP_STATE_UNSPECIFIED"
+	httpEnabled           = "HTTP_ENABLED"
+	httpDisabled          = "HTTP_DISABLED"
+	unspecifiedCertFormat = "UNSPECIFIED_PUBLIC_KEY_CERTIFICATE_FORMAT"
+	x509CertificatePEM    = "X509_CERTIFICATE_PEM"
+)
+
+func resourceCloudiotRegistry() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudiotRegistryCreate,
+		Update: resourceCloudiotRegistryUpdate,
+		Read:   resourceCloudiotRegistryRead,
+		Delete: resourceCloudiotRegistryDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudiotRegistryStateImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: linkDiffSuppress,
+			},
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"event_notification_configs": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				MaxItems: 1,
```

Comment:
If it's a max of one, the name should maybe be the singular "config"?

Reply:
You are right. It sounded strange to me when I read: "Only the first configuration is used." in [eventNotificationConfigs[]](https://cloud.google.com/iot/docs/reference/rest/v1/projects.locations.registries#DeviceRegistry.FIELDS.event_notification_configs). Would it make sense to get rid of the list and map with `pubsub_topic_name`?  I would suggest a field like `event_notification_topic` to keep the topic id.

Reply:
For now I just removed the list schema.